src_c/display.c

```diff
@@ -755,32 +784,45 @@ pg_set_mode(PyObject *self, PyObject *arg, PyObject *kwds)
             else
                 SDL_GL_SetAttribute(SDL_GL_DOUBLEBUFFER, 0);
         }
+
 #pragma PG_WARN(Not setting bpp ?)
 #pragma PG_WARN(Add mode stuff.)
+        int x = SDL_WINDOWPOS_UNDEFINED_DISPLAY(display);
+        int y = SDL_WINDOWPOS_UNDEFINED_DISPLAY(display);
+        int w_1=w;
+        int h_1=h;
+
+        if(win){
+            if (SDL_GetWindowDisplayIndex(win) == display) {
+                SDL_GetWindowPosition(win, &x, &y);
+            }
+            if (!(flags & PGS_OPENGL) !=
+                !(SDL_GetWindowFlags(win) & SDL_WINDOW_OPENGL)){
+                SDL_DestroyWindow(win);
```

Comment:
There may be a memory leak for the placeholder surface here

Reply:
Shouldn't that same memory leak also apply to the fake 1x1 openGL placeholder surface then?